File: output/llama3.1-8b_IO_1/avatar/Python/Java/codeforces_274_A.java
import java.util.*;

public class codeforces_274_A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String[] input1 = scanner.nextLine().split(" ");
        int num = Integer.parseInt(input1[0]);
        int k = Integer.parseInt(input1[1]);

        String[] input2 = scanner.nextLine().split(" ");
        List<Integer> arr = new ArrayList<>();
        for (String s : input2) {
            arr.add(Integer.parseInt(s));
        }

        Set<Integer> different = new HashSet<>();
        Collections.sort(arr);

        int temp = 0;
        for (int x : arr) {
            if ((x % k != 0 || !different.contains(x / k)) && x >= k) {
                different.add(x);
            }
            temp = Math.max(different.size(), temp);
        }

        System.out.println(temp);
    }
}

File: output/llama3.1-8b_IO_1/avatar/Java/Python/codeforces_274_A.py
import sys
from typing import Set

def main():
    n = int(input().split()[0])
    k = int(input())
    arr = []
    set_ = set()

    for _ in range(n):
        arr.append(int(input().split()[0]))

    arr.sort()
    count = 0

    for i in range(n):
        if arr[i] % k != 0:
            count += 1
            set_.add(arr[i])
        elif (arr[i] // k) not in set_:
            count += 1
            set_.add(arr[i])

    print(count)

if __name__ == "__main__":
    main()
